### PR TITLE
Add all.hpp header

### PR DESCRIPTION
**Pull request by jakobod**
This PR adds the `all.hpp` header so a `FindCAF_NET.cmake` script can find the built library.

### CODE REVIEW

File: libcaf_net/caf/net/all.hpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#pragma once
+
+#include "caf/net/actor_proxy_impl.hpp"
```
Comment (josephnoir) [inline libcaf_net/caf/net/all.hpp]: Sounds like it should be private.

File: libcaf_net/caf/net/all.hpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+#include "caf/net/datagram_socket.hpp"
+#include "caf/net/datagram_transport.hpp"
+#include "caf/net/defaults.hpp"
+#include "caf/net/endpoint_manager.hpp"
+#include "caf/net/endpoint_manager_impl.hpp"
```
Comment (josephnoir) [inline libcaf_net/caf/net/all.hpp]: This one as well, maybe?

**Comment by Neverlord** [comment]
> @Neverlord, what do you think?

I think the all header should contain everything in `caf::net` but nothing from nested namespaces such as `net::backend`, `net::basp`, etc. We should probably start auto-generating the `all.hpp` headers.

If the `_impl` classes shouldn't be part of the public API then the also shouldn't live in `caf::net` but in a nested namespace (or `caf::detail`).

**Comment by Neverlord** [comment]
> `FindCAF_NET.cmake` script can find the built library.

One final thought on that: our findCAF script should pick up `caf_net` just fine, e.g., `find_package(CAF COMPONENTS core net)`

**Pull request merged**